test/integration/shoots/creation/create_test.go

```diff
@@ -0,0 +1,224 @@
+	workerZone               = flag.String("worker-zone", "", "zone to use for every worker of the shoot.")
+
+	// ProviderConfigs flags
+	infrastructureProviderConfig = flag.String("infrastructure-provider-config-filepath", "", "filepath to the provider specific infrastructure config")
+	controlPlaneProviderConfig   = flag.String("controlplane-provider-config-filepath", "", "filepath to the provider specific infrastructure config")
+	networkingProviderConfig     = flag.String("networking-provider-config-filepath", "", "filepath to the provider specific infrastructure config")
+
+	// other
+	shootGardenerTest     *ShootGardenerTest
+	testLogger            *logrus.Logger
+	shootYamlPath         = "/example/90-shoot.yaml"
+	trueVar               = true
+	err                   error
+	gardenerTestOperation *GardenerTestOperation
+)
+
+const (
+	CreateAndReconcileTimeout = 7200 * time.Second
+	InitializationTimeout     = 20 * time.Second
+)
+
+func validateFlags() {
+	if !StringSet(*gardenerKubeconfigPath) {
+		Fail("you need to specify the correct path for the gardenerKubeconfigPath")
+	}
+
+	if !FileExists(*gardenerKubeconfigPath) {
+		Fail("gardenerKubeconfigPath path does not exist")
+	}
+
+	if !StringSet(*logLevel) {
+		level := "debug"
+		logLevel = &level
+	}
+
+	if !StringSet(*shootProviderType) {
+		Fail("you need to specify provider type of the shoot")
+	}
+
+	if !StringSet(*projectNamespace) {
+		Fail("you need to specify projectNamespace")
+	}
+
+	if StringSet(*shootMachineImageName) && !StringSet(*shootMachineImageVersion) {
+		Fail("shootMachineImageVersion has to be defined if shootMachineImageName is set")
+	}
+
+	if StringSet(*shootMachineImageVersion) && !StringSet(*shootMachineImageName) {
+		Fail("shootMachineImageName has to be defined if shootMachineImageVersion is set")
+	}
+
+	if !StringSet(*infrastructureProviderConfig) {
+		Fail(fmt.Sprintf("you need to specify the filepath to the infrastructureProviderConfig for the provider '%s'", *shootProviderType))
+	}
+
+	if !FileExists(*infrastructureProviderConfig) {
+		Fail("path to the infrastructureProviderConfig of the Shoot is invalid")
+	}
+
+	if StringSet(*controlPlaneProviderConfig) {
+		if !FileExists(*controlPlaneProviderConfig) {
+			Fail("path to the controlPlaneProviderConfig of the Shoot is invalid")
+		}
+	}
+
+	if StringSet(*networkingProviderConfig) {
+		if !FileExists(*networkingProviderConfig) {
+			Fail("path to the networkingProviderConfig of the Shoot is invalid")
+		}
+	}
+}
+
+var _ = Describe("Shoot Creation testing", func() {
+	CBeforeSuite(func(ctx context.Context) {
+		validateFlags()
+		testLogger = logger.AddWriter(logger.NewLogger(*logLevel), GinkgoWriter)
+
+		shoot := prepareShoot()
+		shootGardenerTest, err = NewShootGardenerTest(*gardenerKubeconfigPath, shoot, testLogger)
+		Expect(err).To(BeNil())
+		shootGardenerTest.SetupShootWorker(workerZone)
```

Comment:
we should handle the error of `SetupShootWorker`

Reply:
Thanks, good catch